### ISSUE TITLE
Invalid location address data is being allowed into the DB

### ISSUE THREAD

**Issue by wsanchez** [burningmantech/ranger-ims-server#64]
Seen on demo server:
```
2016-06-23T17:24:34-0700 [ims.store.sqlite.Storage#critical] Invalid incident (Radial hour must be 1-12, not -1): Incident(number=16,priority=3,summary=None,location=Location(name=None,address=RodGarettAddress(concentric=None,radialHour=-1,radialMinute=None,description=None)),rangers=frozenset([]),incidentTypes=frozenset([]),reportEntries=(),created=datetime.datetime(2016, 6, 24, 0, 24, 32, tzinfo=<ims.tz.FixedOffsetTimeZone object at 0x0000000005954c28>),state=<IncidentState=new>,version=1)
2016-06-23T17:24:34-0700 [ims.service.service.WebService#critical] Request failed

Traceback (most recent call last):
  File "/home/ims/ranger-ims-server/.develop/virtualenv/site-packages/twisted/internet/defer.py", line 1274, in unwindGenerator
    return _inlineCallbacks(None, gen, Deferred())
  File "/home/ims/ranger-ims-server/.develop/virtualenv/site-packages/twisted/internet/defer.py", line 1128, in _inlineCallbacks
    result = g.send(result)
  File "/home/ims/ranger-ims-server/ims/service/klein.py", line 65, in wrapper
    response = yield f(self, request, *args, **kwargs)
  File "/home/ims/ranger-ims-server/.develop/virtualenv/site-packages/twisted/internet/defer.py", line 1274, in unwindGenerator
    return _inlineCallbacks(None, gen, Deferred())
--- <exception caught here> ---
  File "/home/ims/ranger-ims-server/.develop/virtualenv/site-packages/twisted/internet/defer.py", line 1128, in _inlineCallbacks
    result = g.send(result)
  File "/home/ims/ranger-ims-server/ims/service/json.py", line 180, in readIncidentResource
    incident = self.storage.incident(event, number)
  File "/home/ims/ranger-ims-server/ims/store/sqlite.py", line 308, in incident
    incident.validate()
  File "/home/ims/ranger-ims-server/ims/data/model.py", line 326, in validate
    "location", self.location, Location, optional=True, recurse=True
  File "/home/ims/ranger-ims-server/ims/data/model.py", line 135, in _validateIsInstance
    obj.validate()
  File "/home/ims/ranger-ims-server/ims/data/model.py", line 630, in validate
    "address", self.address, Address, optional=True, recurse=True
  File "/home/ims/ranger-ims-server/ims/data/model.py", line 135, in _validateIsInstance
    obj.validate()
  File "/home/ims/ranger-ims-server/ims/data/model.py", line 862, in validate
    .format(self.radialHour)
ims.data.model.InvalidDataError: Radial hour must be 1-12, not -1
```

As noted in #63: "I set the hour to '1' and then set it back to the blank hour."

**Comment by corprew** [comment]
same issue seems to happen with the street name. (repro: set to 'esplanade' then back to '')

**Comment by wsanchez** [comment]
OK this is bad because it looks like I allowed bad data to actually get persisted, and not the DB is vending bad data which the client except it poops in the server on the way out which is why #65 happens.

**Comment by wsanchez** [comment]
6d1a059babae5e5983d4f2e63455ecec9fb21f48 adds a constraint so the DB will disallow invalid concentric streets.

**Comment by wsanchez** [comment]
We still allow bogus data for radial street components, however.

The reason this is allowed right now is that the code for editing an incident (versus creating a new one) applies the changes piecemeal to the DB, which fits the UI model on the web interface and is efficient, but doesn't instantiate the whole incident and validate it.

**Comment by wsanchez** [comment]
c251c9a7422af6c0cc66ef0255a66613134c8451 prevents bad data from going into the DB

**Comment by wsanchez** [comment]
Fixed: 6a54c55afcdb4b83d5ab9e1aab173ca313201f7b
